src/collections/map.rs

```diff
@@ -64,13 +126,14 @@ where
     S: Read,
```

Comment:
Now that we use `MapKey`, can't we get rid of the `Eq`/`Hash`/`Ord` bounds (and maybe `Clone`) for `K` in all these impl blocks? (For `Hash` this would involve implementing `Hash` for `MapKey`, by writing the encoded bytes to the hasher).

Reply:
Got rid of a lot of the bounds here and moved a few methods to less restrictive blocks.